### PR TITLE
Implement authentification

### PR DESCRIPTION
**Pull request by D3nn7**
- add appwrite backend functions
- add auth using jwt and email
- add register 
- add types and recoil states
- add Page component for routing and user state management
- add basic components
- add basic api for times and categories
- remove .env file
- use Nextjs config file instead of .env
- edit README
- remove second README

### CODE REVIEW

**Comment by D3nn7** [comment]
I need to implement the account delete function and password reset before we can merge this.

**Comment by D3nn7** [comment]
I have implemented the Password reset and change now. 
Please review.

**Comment by D3nn7** [comment]
I'll fix the code smells first.

File: worktime-tracker-dev/pages/account/overview.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+        fetchData();
+    }, [user, userProfilePicture]);
+
+    return (
+        <Page isSecurePage={true} isLoading={loading}>
+            <section className="container h-screen max-h-screen px-3 max-w-xl mx-auto flex flex-col">
+                <div className="my-auto p-16 rounded-lg text-center">
+                    <div className="font-bold text-3xl md:text-5xl lg:text-6xl">
+                        Hej {user?.name}
```
Comment (Jonas22rr) [inline worktime-tracker-dev/pages/account/overview.tsx:32]: perhaps take not optional user name -> when user.name not exist got to login?!

File: worktime-tracker-dev/pages/account/register.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import Link from 'next/link'
+import { useRouter } from "next/router";
+import Alert from "../../components/page/alert";
+import { appwrite, userState } from "../../store/global";
+import { useRecoilState } from 'recoil';
+import { User } from '../../store/types';
+import Page from "../../components/page/page";
+
+const Register = () => {
+    const [name, setName] = useState("");
```
Comment (Jonas22rr) [inline worktime-tracker-dev/pages/account/register.tsx:11]: types on useState

File: worktime-tracker-dev/pages/account/register.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [alert, setAlert] = useState("");
+    const [ , setUser] = useRecoilState(userState);
+    const router = useRouter();
+
+    const register = async (e: FormEvent<EventTarget>) => {
```
Comment (Jonas22rr) [inline worktime-tracker-dev/pages/account/register.tsx:18]: e -> event (whole file)

File: worktime-tracker-dev/pages/account/reset.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import { GetServerSideProps } from "next";
+import { useRouter } from "next/router";
+import { FormEvent, useEffect, useState } from "react";
+import Alert from "../../components/page/alert";
+import Page from "../../components/page/page"
+import { appwrite } from "../../store/global";
+import { AppwriteErrorType } from "../../utils/appwriteResponse";
+
+export default function Reset({ userId, secret, expire } : { userId?: string, secret?: string, expire?: string }) {
+    const [alert, setAlert] = useState("");
```
Comment (Jonas22rr) [inline worktime-tracker-dev/pages/account/reset.tsx:11]: types

File: worktime-tracker-dev/pages/account/reset.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    const [email, setEmail] = useState("");
+
+    const [password, setPassword] = useState("");
+    const [passwordRepeat, setPasswordRepeat] = useState("");
+
+    const [resetProcess, setResetProcess] = useState(false);
+    const router = useRouter();
+
+    const resetPassword = async (e: FormEvent<EventTarget>) => {
```
Comment (Jonas22rr) [inline worktime-tracker-dev/pages/account/reset.tsx:20]: e -> event

File: worktime-tracker-dev/pages/account/verify.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import { useEffect, useState } from "react";
+import { useRecoilState } from "recoil";
+import Alert from "../../components/page/alert";
+import Page from "../../components/page/page"
+import { appwrite, userState } from "../../store/global";
+import { AppwriteErrorType } from "../../utils/appwriteResponse";
+
+export default function Verify({ userId, secret} : { userId?: string, secret?: string }) {
+	const [user] = useRecoilState(userState);
+	const [verified, setVerified] = useState(false);
```
Comment (Jonas22rr) [inline worktime-tracker-dev/pages/account/verify.tsx:13]: types

**Review by Jonas22rr — APPROVED** [review]

File: worktime-tracker-dev/pages/account/overview.tsx (hunk 1 above)
Comment (D3nn7) [reply worktime-tracker-dev/pages/account/overview.tsx:32]: The '?' is because the state loads async.

**Comment by D3nn7** [comment]
I have changed the suggest points pls review.

**Pull request merged**